lib/ansible/modules/system/macos_pref.py

```diff
@@ -0,0 +1,1079 @@
+            if self.type:
+                self.value = self._cast_type(self.value, self.type)
+            else:
+                self.value = self._auto_cast_type(self.value)
+
+            # For state=merge, when base is array convert single item to a list.
+            if self.merge and isinstance(self.current_value, list) and not isinstance(self.value, list):
+                self.value = [self.value]
+
+            # Check if there's a type mismatch.
+            if (self.current_value is not None and
+                    not equivalent_types(self.current_value, self.value)):
+                raise MacOSPrefException(
+                    'New value type does not match current value type for key '
+                    '{0} ({1!r} {2} -> {3!r} {4}).'
+                    .format(
+                        self.key, self.value, type(self.value),
+                        self.current_value, type(self.current_value)
+                    )
+                )
+
+            # Check for changes and abort if ther are none
+            if self.merge:
+                if self.current_value is not None:
+                    # compare arrays
+                    if isinstance(self.current_value, list):
+                        new_items = [
+                            item for item in self.value
+                            if item not in self.current_value
+                        ]
+                        if not new_items:
+                            return
+                    # compare dicts
+                    else:
+                        new_items = [
+                            key for key, val in self.value.items()
+                            if key not in self.current_value
+                        ]
+                        if not new_items:
+                            return
+            elif self.current_value == self.value:
+                return
+
+            self.changed = True
+
+            if self.check_mode:
+                return
+
+            # Change/Create/Set given key/value for domain in defaults.
+            self.prefs.write(self.key, self.value, self.merge)
+
+
+def main():
+    module = AnsibleModule(
+        argument_spec=dict(
+            domain=dict(
+                default='NSGlobalDomain',
+                type='str',
+                required=False
+            ),
+            user=dict(
+                default='currentUser',
+                choices=[
+                    'anyUser',
+                    'currentUser'
+                ],
+                required=False
+            ),
+            host=dict(
+                default='anyHost',
+                choices=[
+                    'anyHost',
+                    'currentHost'
+                ],
+                required=False
+            ),
+            key=dict(
+                type='str',
+                required=True
+            ),
+            type=dict(
+                default=None,
```

Comment:
Default None values do not have to be added, because that is the default.
It's implicit that unset parameters always return None (and that's one way to check for a value)